protocol/vdp_protocol.yaml

```diff
@@ -5,273 +5,570 @@ title: VDP Protocol
 type: object
 description: VDP Protocol structs
 additionalProperties: true
-anyOf:
-  - required:
-      - classification
-  - required:
-      - detection
-  - required:
-      - keypoint
-  - required:
-      - ocr
-  - required:
-      - instance_segmentation
-  - required:
-      - semantic_segmentation
-  - required:
-      - text_to_image
-  - required:
-      - text_generation
-  - required:
-      - unspecified
+oneOf:
+- required:
+  - classification
+- required:
+  - detection
+- required:
+  - keypoint
+- required:
+  - ocr
+- required:
+  - instance_segmentation
+- required:
+  - semantic_segmentation
+- required:
+  - text_to_image
+- required:
+  - text_generation
+- required:
+  - unspecified
 properties:
   classification:
-    description: "Classify an image into pre-defined categories"
+    description: Classify an image into pre-defined categories
     "$ref": "#/definitions/Classification"
   detection:
-    description: "Detect and localise multiple objects"
+    description: Detect and localise multiple objects
     "$ref": "#/definitions/Detection"
   keypoint:
-    description: "Detect and localise keypoints of multiple objects"
+    description: Detect and localise keypoints of multiple objects
     "$ref": "#/definitions/Keypoint"
   ocr:
-    description: "Detect, localise and recognise texts"
+    description: Detect, localise and recognise texts
     "$ref": "#/definitions/Ocr"
   instance_segmentation:
-    description: "Detect, localise and delineate multiple objects"
+    description: Detect, localise and delineate multiple objects
     "$ref": "#/definitions/InstanceSegmentation"
   semantic_segmentation:
-    description: "Classify image pixels into predefined categories"
+    description: Classify image pixels into predefined categories
     "$ref": "#/definitions/SemanticSegmentation"
   text_to_image:
-    description: "Generates images from input text"
+    description: Generates images from input text
     "$ref": "#/definitions/TextToImage"
   text_generation:
-    description: "Generate new texts from input text"
+    description: Generate new texts from input text
     "$ref": "#/definitions/TextGeneration"
   unspecified:
-    description: "Unspecified task with output in the free form"
+    description: Unspecified task with output in the free form
     "$ref": "#/definitions/Unspecified"
 definitions:
   Classification:
     type: object
     additionalProperties: false
-    required:
-      - category
-      - score
-    properties:
-      category:
-        description: "The predicted category of the input"
-        type: string
-      score:
-        description: "The confidence score of the predicted category of the input"
-        type: number
-  Detection:
-    type: object
-    additionalProperties: false
-    required:
-      - objects
+    oneOf:
+    - required:
+      - input
```

Comment:
I suggest we use plural `inputs` and `outputs`

Reply:
changed